caas/kubernetes/provider/application/constraints.go

```diff
@@ -266,6 +273,165 @@ func processPodAffinity(pod *core.PodSpec, affinityLabels map[string]string) err
+			continue
+		}
+
+		key = strings.TrimPrefix(key, topologySpreadPrefix)
+		if !validKeys.Contains(key) {
+			return errors.Errorf("invalid topology spread constraint key %q", key)
+		}
+		if key == topologySpreadMaxSkew {
+			val, err := strconv.Atoi(value)
+			if err != nil {
+				return errors.Errorf("invalid value %q for topology spread max skew", value)
+			}
+			if val < 1 {
+				return errors.Errorf("invalid value, maxSkew must be greater or equal to 1")
+			}
+		}
+		topologySpreadValues[key] = value
+	}
+	if len(topologySpreadValues) == 0 {
+		return nil
+	}
+	_, present := topologySpreadValues[topologyKeyTag]
+	if !present {
+		return errors.Errorf("topology-key not set for topology spread constraints: %v", affinityLabels)
+	}
+
+	var appNameLabel string = "app.kubernetes.io/name"
+
+	updateTopologyTerm := func(topologyTerms *core.TopologySpreadConstraint, tags map[string]string) {
+		// Sort for stable ordering.
+		var keys []string
+		for k := range tags {
+			keys = append(keys, k)
+		}
+		sort.Strings(keys)
+		var (
+			labelSelector    v1.LabelSelector
+			topologyKey      string = "kubernetes.io/zone"
+			maxSkew          int    = 1
+			minDomains       int    = 3
+			nodeTaintsPolicy *core.NodeInclusionPolicy
+		)
+
+		for _, key := range keys {
+
+			if key == topologyKeyTag {
+				topologyKey = tags[topologyKeyTag]
+				continue
+			}
+			if key == topologySpreadMinDomains {
+				domains, ok := strconv.Atoi(tags[topologySpreadMinDomains])
+				if ok != nil || domains < 2 {
+					minDomains = 2
+				} else {
+					minDomains = domains
+				}
+				continue
+			}
+			if key == topologySpreadMaxSkew {
+				skew, ok := strconv.Atoi(tags[topologySpreadMaxSkew])
+				if ok != nil || skew < 1 {
+					maxSkew = 1
+				} else {
+					maxSkew = skew
+				}
+				continue
+			}
+
+			if key == topologySpreadNodeTaintPolicy {
+				taintPolicy, err := strconv.ParseBool(tags[key])
+				if err != nil {
+					taintPolicy = true
+				}
+				if taintPolicy {
+					honorPolicy := core.NodeInclusionPolicy("Honor")
+					nodeTaintsPolicy = &honorPolicy
+				} else {
+					ignorePolicy := core.NodeInclusionPolicy("Ignore")
+					nodeTaintsPolicy = &ignorePolicy
+				}
+				continue
+			}
+
+			if key == appNameLabel {
+				// Nothing to do, we will add this key anyways
+				continue
+			}
+
+			allValues := strings.Split(tags[key], "|")
+			for i, v := range allValues {
+				allValues[i] = strings.Trim(v, " ")
+			}
+			op := v1.LabelSelectorOpIn
+			if strings.HasPrefix(key, "^") {
+				key = key[1:]
+				op = v1.LabelSelectorOpNotIn
+			}
+			labelSelector.MatchExpressions = append(labelSelector.MatchExpressions, v1.LabelSelectorRequirement{
+				Key:      key,
+				Operator: op,
+				Values:   allValues,
+			})
+		}
+		if topologyKey != "" {
+			topologyTerms.TopologyKey = topologyKey
+		}
+		topologyTerms.WhenUnsatisfiable = core.DoNotSchedule
+		topologyTerms.MaxSkew = int32(maxSkew)
+		minimumDomains := int32(minDomains)
```

Comment:
Define the vars as in32 in the first place so we can avoid the cast here, it's cleaner.

Reply:
Not sure how can I do it, as I am using *Atoi* right before.